sample/gofc.go

```diff
@@ -70,10 +73,21 @@ func (c *SampleController) HandleAggregateStatsReply(msg *ofp13.OfpMultipartRepl
 }
 
 func main() {
-	// regist app
-	ofc := NewSampleController()
-	gofc.GetAppManager().RegistApplication(ofc)
+	// register app
+	controller:=gofc.NewOFController()
+	controller.RegisterHandler(NewSampleController())
 
 	// start server
-	gofc.ServerLoop(gofc.DEFAULT_PORT)
+
+	l,err:=net.ListenTCP("tcp4", &net.TCPAddr{IP:net.ParseIP("127.0.0.1"), Port:gofc.DEFAULT_PORT})
+	if err != nil{
+		panic(err)
+	}
+	go func () {
+		if err:=controller.Serve(l); err !=nil {
+			panic(err)
+		}
+	}()
+	time.Sleep(time.Second*15)
+	err = controller.Shutdown(context.Background())
```

Comment:
may be it would be better to use defer or something like the following:
```suggestion
// setup signal catching
        sigs := make(chan os.Signal, 1)
        // catch all signals since not explicitly listing
        signal.Notify(sigs)
        //signal.Notify(sigs,syscall.SIGQUIT)
        // method invoked upon seeing signal
        go func() {
          s := <-sigs
          log.Printf("RECEIVED SIGNAL: %s",s)
          err = controller.Shutdown(context.Background())
          os.Exit(1)
        }()	
```
https://fabianlee.org/2017/05/21/golang-running-a-go-binary-as-a-systemd-service-on-ubuntu-16-04/

Reply:
Fixed.

Reply:
That was fixed with use one receiver buffer and copy data from slice to new buffer before send in to receive channel.